api/python/quilt3/data_transfer.py

```diff
@@ -168,20 +168,22 @@ def register_signals(self, s3_client):
                 event_name, signal_transferring,
                 unique_id='datatransfer-transferring')
 
-    def _build_standard_client(self):
-        boto_session = self.get_boto_session()
-
-        config = None
-        if boto_session.get_credentials() is None:
-            config = Config(signature_version=UNSIGNED)
+    def _build_client(self, get_config):
+        session = self.get_boto_session()
+        return session.client('s3', config=get_config(session))
 
-        s3_client = boto_session.client('s3', config=config)
+    def _build_standard_client(self):
+        s3_client = self._build_client(
+            lambda session:
+                Config(signature_version=UNSIGNED)
+                if session.get_credentials() is None
+                else None
```

Comment:
I was wondering if `session.client('s3', config=None)` would work (it does) and I think it's worth linking the docs in a comment: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client

Reply:
Don't understand, could you elaborate on that?

Reply:
Yes. The reader knows that config=None can happen but without docs can't know if that call is legal or will crash boto